frontends/web/src/components/devices/bitbox02bootloader/bitbox02bootloader.tsx

```diff
@@ -14,165 +15,109 @@
-              style={{ textDecoration: 'underline', cursor: 'pointer' }} >
-              {t('bb02Bootloader.flipscreen')}
-            </a>
-          </div>
-          <hr/>
-          <details>
-            <summary>
-              {t('bb02Bootloader.advanced.label')}
-            </summary>
-            <div>
-              <br />
-              <ToggleShowFirmwareHash deviceID={deviceID} />
-            </div>
-          </details>
+        <div className="box large">
+          <progress value={value} max="100">{value}%</progress>
+          <p style={{ marginBottom: 0 }}>
+            {t('bootloader.progress', {
+              progress: value.toString(),
+              context: (versionInfo.erased ? 'install' : ''),
+            })}
+          </p>
         </div>
       );
     }
-    return (
-      <CenteredContent>
-        {getDarkmode() ? <BitBox02Inverted /> : <BitBox02 />}
-        {status.errMsg && (
-          <Status type="warning">{status.errMsg}</Status>
+  } else {
+    contents = (
+      <div className="box large" style={{ minHeight: 390 }}>
+        {versionInfo.erased && (
+          <div>
+            <h2>{t('welcome.title')}</h2>
+            <h3 className="subTitle">{t('welcome.getStarted')}</h3>
+          </div>
         )}
-        {contents}
-      </CenteredContent>
+        <div className="buttons">
+          { versionInfo.canUpgrade ? (
+            <Button
+              primary
+              onClick={() => bitbox02BootloaderAPI.upgradeFirmware(deviceID)}>
+              {t('bootloader.button', { context: (versionInfo.erased ? 'install' : '') })}
+            </Button>
+          ) : null }
+          { !versionInfo.erased && (
+            <Button
+              secondary
+              onClick={() => bitbox02BootloaderAPI.reboot(deviceID)}>
+              {t('bb02Bootloader.abort', { context: !versionInfo.canUpgrade ? 'noUpgrade' : '' })}
+            </Button>
+          )}
+        </div>
+        <div className="flex flex-center" style={{ marginTop: 32 }}>
+          {t('bb02Bootloader.orientation')}&nbsp;
+          <a
+            onClick={() => bitbox02BootloaderAPI.screenRotate(deviceID)}
+            style={{ textDecoration: 'underline', cursor: 'pointer' }} >
+            {t('bb02Bootloader.flipscreen')}
+          </a>
+        </div>
+        <hr/>
+        <details>
+          <summary>
+            {t('bb02Bootloader.advanced.label')}
+          </summary>
+          <div>
+            <br />
+            <ToggleShowFirmwareHash deviceID={deviceID} />
+          </div>
+        </details>
+      </div>
     );
   }
-}
-
-const loadHOC = load<LoadedProps, BitBox02BootloaderProps & TranslateProps>(({ deviceID }) => ({ versionInfo: 'devices/bitbox02-bootloader/' + deviceID + '/version-info' }))(BitBox02Bootloader);
-const HOC = translate()(loadHOC);
-export { HOC as BitBox02Bootloader };
+  return (
+    <CenteredContent>
+      {getDarkmode() ? <BitBox02Inverted /> : <BitBox02 />}
```

Comment:
nit: with functional component we could now use darkmode hook.

```js
import { useDarkmode } from '../hooks/darkmode';

const { isDarkMode } = useDarkmode();
```

Reply:
great! done